pkg/chunk/series_store.go

```diff
@@ -440,16 +445,23 @@ func (c *seriesStore) PutOne(ctx context.Context, from, through model.Time, chun
 			return err
 		}
 	} else {
-		err := c.storage.PutChunks(ctx, chunks)
-		if err != nil {
-			return err
+		// chunk not found, write it.
+		if len(found) == 0 {
```

Comment:
Can we introduce new bool variables `writeChunk` and `writeIndex`, and set them at the beginning based on `len(found)` and `c.cfg.DisableIndexDeduplication`? I think it would make code slightly easier to follow. WDYT?

Reply:
Agree on this.

Reply:
I have added just the `writeChunk` variable because writing chunk is optional while the index is not.